src/lib/wallet.js

```diff
@@ -38,9 +38,30 @@ export default async function initSdk() {
 
   store.commit('setNodeStatus', NODE_STATUS_CONNECTING);
   try {
+    const { getMiddleware, initMiddleware } = useMiddleware({ store });
+    store.watch(
+      (state, getters) => getters.activeNetwork,
+      async (network, oldNetwork) => {
+        if (isEqual(network, oldNetwork)) return;
+        try {
+          if (store.state.nodeStatus === NODE_STATUS_ERROR && !store.getters['sdkPlugin/sdk']) {
+            await initSdk();
+            return;
+          }
+          store.commit('setNodeStatus', NODE_STATUS_CONNECTING);
+          await Promise.all([
+            network.middlewareUrl !== oldNetwork.middlewareUrl ? initMiddleware() : null,
+            network.url !== oldNetwork.url ? store.dispatch('sdkPlugin/changeNode', network) : null,
+          ]);
```

Comment:
`network.middlewareUrl !== oldNetwork.middlewareUrl` & `network.url !== oldNetwork.url` are unnecessary given the initial condition `if (isEqual(network, oldNetwork)) return;` in this watcher.

Reply:
User cancreate a new network using same url, but different middleware url.

Reply:
Oh yeah. You're right!